superset/views/core.py

```diff
@@ -842,6 +842,16 @@ def request_access(self):
                 .one()
             )
             datasources.add(datasource)
+
+        has_access = all(
+            (
+                datasource and self.datasource_access(datasource)
+                for datasource in datasources
+            ),
```

Comment:
Watch out here, I'm pretty sure this comma changes the structure from passing a tuple to `all` to, instead, passing it a tuple of one element that contains a tuple.

Reply:
It's a pretty common gotcha in python, `v = 1,` makes `v` a tuple